react-accordion/my-app/src/Accordion.tsx

```diff
@@ -0,0 +1,43 @@
+import './Accordion.css';
+import { useState } from 'react';
+
+type Prop = {
+  topics: {
+    id: number;
+    title: string;
+    content: string;
+  }[];
+};
+
+export function Accordion({ topics }: Prop) {
+  const [currentID, setCurrentID] = useState(0);
+  const topicComponents = topics.map((topic) => {
+    return (
+      <div key={topic.id}>
+        <TopicCard topic={topic} currentID={currentID} onClick={setCurrentID} />
+      </div>
+    );
+  });
+  return <>{topicComponents}</>;
+}
+
+type CardProp = {
+  topic: {
+    id: number;
+    title: string;
+    content: string;
+  };
+  currentID: number;
+  onClick: (arg0: number) => void;
+};
+
+function TopicCard({ topic, currentID, onClick }: CardProp) {
+  return (
+    <>
+      <button onClick={() => onClick(topic.id === currentID ? 0 : topic.id)}>
```

Comment:
Minor, but I think the logic for setting the currentID to 0 should be in the parent, not here. This can just report that it was clicked on and let the parent decide what to do about it.

To do it that way, you would probably want to send a prop like "isOpen" instead of the current ID.